packages/cli/src/commands/stream/deterministic.ts

```diff
@@ -0,0 +1,87 @@
+import { flags } from '@oclif/command'
+import { TileDocument } from '@ceramicnetwork/stream-tile'
+import type { TileMetadataArgs } from '@ceramicnetwork/stream-tile'
+
+import { Command } from '../../command'
+import type { CommandFlags } from '../../command'
+import { parseControllers } from '../../utils'
+
+type Flags = CommandFlags & {
+  controller?: string
+  metadata?: TileMetadataArgs
+}
+
+export default class Deterministic extends Command<
+  Flags,
+  {
+    content: string
+    schemaId?: string
+  }
+> {
+  static description = 'Create a new deterministic Stream'
+
+  static args = [
+    {
```

Comment:
This is super confusing. We have the `schema` as argument, `controller` as explicit flag and another `metadata` flag that could also include these?
I think we should have `metadata` as single required argument to make it more explicit, and no other argument/flag that could create confusion as to what value is used.

Reply:
Agreed, I did however decide to keep metadata as optional & if there is no metadata we create it containing the controller & will also set the controller to the user DID if nothing else is provided.

Reply:
The `metadata` should be mandatory, that's what the determinism is based on.